hypothesis-python/src/hypothesis/internal/conjecture/shrinker.py

```diff
@@ -686,37 +611,35 @@ def pass_to_descendant(self, ancestor_id, descendant_id):
         as possible.
         """
 
-        ancestor = self.example_for_stable_identifier(ancestor_id)
-        descendant = self.example_for_stable_identifier(descendant_id)
-        if descendant is None:
-            return
-        assert ancestor is not None
+        label = chooser.choose(
+            self.distinct_labels, lambda l: len(self.examples_by_label[l]) >= 1
+        )
+
+        ls = self.examples_by_label[label]
+
+        i = chooser.choose(hrange(len(ls)))
```

Comment:
Similarly, this can be `hrange(len(ls) - 1)`, since a label's final example can't have any same-label descendants.